automation/service/workflow_converter.go

```diff
@@ -506,6 +512,92 @@ func (svc workflowConverter) convContinueStep() (wfexec.Step, error) {
 
 }
 
+// creates workflow-executor step
+//
+// Expects max ONE outgoing paths and ONE incoming path
+//
+//
+func (svc workflowConverter) convExecWorkflowStep(wf *types.Workflow, s *types.WorkflowStep) (_ wfexec.Step, err error) {
+	const (
+		// sub-workflow's scope
+		// expecting expr.Vars
+		argNameScope = "scope"
+
+		// run workflow from a specific step
+		// must be an orphan step
+		argNameStepID = "stepID"
+
+		// workflow id or handle
+		argNameWorkflowID     = "workflowID"
+		argNameWorkflowHandle = "workflowHandle"
+	)
+
+	return wfexec.NewGenericStep(func(ctx context.Context, r *wfexec.ExecRequest) (resp wfexec.ExecResponse, err error) {
+		var (
+			p = types.WorkflowExecParams{
+				// sub-workflows should not be executed asynchronously
+				// might lead to uncontrolled resource consumption
+				Async: false,
+
+				// @todo figure out how check if trace is enabled
+				Trace: false,
+
+				// always wait for the sub-workflow even when deferred
+				Wait: true,
+
+				Input: expr.EmptyVars(),
+
+				// who's calling sub-workflow?
+				CallerSessionID:  r.SessionID,
+				CallerStepID:     s.ID,
+				CallerWorkflowID: wf.ID,
+			}
+
+			result *expr.Vars
+
+			workflowID     uint64
+			workflowHandle string
+		)
+
+		if ap, err := processArguments(ctx, s.Arguments, r.Scope); err != nil {
+			return nil, err
+		} else {
+			ap.string(argNameWorkflowHandle, &workflowHandle)
+			ap.uint64(argNameWorkflowID, &workflowID)
+
+			// optional stepID, follows login in Exec when 0
+			ap.uint64(argNameStepID, &p.StepID)
+
+			// scope for the sub-workflow
+			ap.vars(argNameScope, p.Input)
+		}
+
+		if workflowID > 0 && workflowHandle != "" {
+			return nil, fmt.Errorf("cannot provide both workflowID and workflowHandle")
+		}
+
+		if workflowID == 0 {
+			if workflowHandle == "" {
+				return nil, fmt.Errorf("workflowID or workflowHandle must be provided")
+			}
+
+			workflowID = svc.graphs.handleToID(workflowHandle)
+		}
```

Comment:
This might be a bit easier on the eyes (I prefer this form)

```go
		if workflowID == 0 {
			workflowID = svc.graphs.handleToID(workflowHandle)
               }
		if workflowID == 0 {
                        return nil, fmt.Errorf("workflowID or workflowHandle must be provided")
		}
```